src/disco/topo/fd_topo.c

```diff
@@ -318,14 +314,14 @@ fd_topo_workspace_fill( fd_topo_t *      topo,
       fd_topo_tile_t * tile = &topo->tiles[ i ];
       if( FD_LIKELY( tile->wksp_id!=wksp->id ) ) continue;
 
-      fd_tile_config_t * config = fd_topo_tile_to_config( tile );
-      if( FD_LIKELY( config->scratch_align ) ) {
+      ulong align = tile_align( tile );
+      if( FD_LIKELY( align ) ) {
         ulong current_wksp_offset = fd_ulong_align_up( fd_wksp_private_data_off( part_max ), fd_topo_workspace_align() ) + scratch_top;
 
-        ulong desired_scratch_align = config->scratch_align();
+        ulong desired_scratch_align = tile_align( tile );
         ulong pad_to_align = desired_scratch_align - (current_wksp_offset % desired_scratch_align);
```

Comment:
Can you switch this to `fd_ulong_align_up`? If `current_wksp_offset` is a multiple of `desired_scratch_align`, then this adds `desired_scratch_align` instead of 0. That's fine, but not sure it is intentional.